packages/openapi-generator/src/generator.ts

```diff
@@ -29,56 +33,30 @@ const logger = createLogger('openapi-generator');
 export async function generate(options: GeneratorOptions): Promise<void> {
   options.serviceMapping =
     options.serviceMapping ||
-    resolve(options.inputDir.toString(), 'service-mapping.json');
+    resolve(options.input.toString(), 'service-mapping.json');
 
   if (options.clearOutputDir) {
     await rmdir(options.outputDir, { recursive: true });
   }
 
-  // TODO: should be recursive
-  const inputFilePaths = (await readdir(options.inputDir)).map(fileName =>
-    resolve(options.inputDir, fileName)
-  );
-
   const vdmMapping = readServiceMapping(options);
 
-  inputFilePaths.forEach(async filePath => {
-    const serviceName = parseServiceName(filePath);
-    // TODO: get kebapcase unique directory name
-    const serviceDir = resolve(options.outputDir, serviceName);
-    let openApiDocument;
-    try {
-      openApiDocument = await convertOpenApiSpec(filePath);
-    } catch (err) {
-      logger.error(
-        `Could not convert document at ${filePath} to the format needed for parsing and generation. Skipping service generation.`
+  const uniqueNameGenerator = new UniqueNameGenerator('-');
+
+  if ((await lstat(options.input)).isFile()) {
+    const inputFilePath = options.input;
+    generateFromFile(inputFilePath, options, vdmMapping, uniqueNameGenerator);
+  } else {
+    const inputFilePaths = await getInputFilePaths(options.input);
+    inputFilePaths.forEach(async filePath => {
+      await generateFromFile(
+        filePath,
+        options,
+        vdmMapping,
+        uniqueNameGenerator
```

Comment:
You can reduce the code a bit by moving the check whether it is a file to the `getInputFilePaths` function (see below).
```suggestion
  getInputFilePaths(options.input).forEach(async filePath => {
      await generateFromFile(
        filePath,
        options,
        vdmMapping,
        uniqueNameGenerator
      )
```

Reply:
Also, you pointed out very correctly, that forEach executes the functions asynchronously. Therefore the generated names could differ between multiple runs.

There are two possibilities to solve that (and you could use reduce, but nah) :
1. Execute this in a `for ... of` loop as you did before.
2. Generate the name before calling `generateFromFile` and pass the name to `generateFromFile` instead of the uniqueNameGenerator.

There are pros and cons for both options, but I think for now I would go with 1. and add a comment, so that nobody accidentally removes this (it is easy to miss this).